File: app/src/main/java/com/kota/Bahamut/Pages/Model/BoardPageItem.java
package com.kota.Bahamut.Pages.Model;

import android.text.SpannableString;

import com.kota.Bahamut.ListPage.TelnetListPageItem;
import com.kota.Telnet.Model.TelnetRow;

import java.util.Stack;

public class BoardPageItem extends TelnetListPageItem {
    private static final Stack<BoardPageItem> _pool = new Stack<>();
    public String Author = null;
    public String Date = null;
    public int GY = 0;
    public int Size = 0;
    public String Title = null;
    public boolean isMarked = false;
    public boolean isRead = false;
    public boolean isReply = false;

    private BoardPageItem() {
    }

    public static void release() {
        synchronized (_pool) {
            _pool.clear();
        }
    }

    public static void recycle(BoardPageItem item) {
        synchronized (_pool) {
            _pool.push(item);
        }
    }

    public static BoardPageItem create() {
        BoardPageItem item = null;
        synchronized (_pool) {
            if (_pool.size() > 0) {
                item = _pool.pop();
            }
        }
        if (item == null) {
            return new BoardPageItem();
        }
        return item;
    }

    public void clear() {
        super.clear();
        this.Size = 0;
        this.isRead = false;
        this.isMarked = false;
        this.isReply = false;
        this.GY = 0;
        this.Date = null;
        this.Author = null;
        this.Title = null;
    }

    protected void finalize() throws Throwable {
        super.finalize();
    }

    public void set(TelnetListPageItem item) {
        super.set(item);
        if (item != null) {
            BoardPageItem article_item = (BoardPageItem) item;
            this.Size = article_item.Size;
            this.Date = article_item.Date;
            this.Author = article_item.Author;
            this.isRead = article_item.isRead;
            this.isMarked = article_item.isMarked;
            this.isReply = article_item.isReply;
            this.GY = article_item.GY;
            this.Title = article_item.Title;
        }
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/model/BoardPageItem.kt
package com.kota.Bahamut.pages.model

import com.kota.Bahamut.listPage.TelnetListPageItem
import java.util.Stack

class BoardPageItem private constructor() : TelnetListPageItem() {
    var author: String? = null
    var date: String? = null
    var gy: Int = 0
    var size: Int = 0
    var title: String? = null
    var isMarked: Boolean = false
    var isRead: Boolean = false
    var isReply: Boolean = false

    override fun clear() {
        super.clear()
        this.size = 0
        this.isRead = false
        this.isMarked = false
        this.isReply = false
        this.gy = 0
        this.date = null
        this.author = null
        this.title = null
    }

    override fun set(aData: TelnetListPageItem) {
        super.set(aData)
        val articleItem = aData as BoardPageItem
        this.size = articleItem.size
        this.date = articleItem.date
        this.author = articleItem.author
        this.isRead = articleItem.isRead
        this.isMarked = articleItem.isMarked
        this.isReply = articleItem.isReply
        this.gy = articleItem.gy
        this.title = articleItem.title
    }

    companion object {
        private val _pool = Stack<BoardPageItem?>()
        @JvmStatic
        fun release() {
            synchronized(_pool) {
                _pool.clear()
            }
        }

        fun recycle(item: BoardPageItem?) {
            synchronized(_pool) {
                _pool.push(item)
            }
        }

        fun create(): BoardPageItem {
            var item: BoardPageItem? = null
            synchronized(_pool) {
                if (_pool.isNotEmpty()) {
                    item = _pool.pop()
                }
            }
            if (item == null) {
                return BoardPageItem()
            }
            return item
        }
    }
}
